lib/semantics/check-omp-structure.cc

```diff
@@ -30,52 +31,51 @@ bool OmpStructureChecker::HasInvalidCloselyNestedRegion(
   return false;
 }
 
-void OmpStructureChecker::CheckAllowed(const OmpClauseEnum &type) {
+void OmpStructureChecker::CheckAllowed(const OmpClause &type) {
   if (ompContext_.empty()) {
     return;
   }
-  bool f{true};
-  if (!GetOmpContext().allowedClauses.test(type)) {
-    context_.Say(GetOmpContext().currentClauseSource,
-        "'%s' clause is not allowed on the %s directive"_err_en_US,
-        parser::ToUpperCaseLetters(EnumToString(type)),
+  if ((!GetContext().allowedClauses.empty() &&
+          !GetContext().allowedClauses.test(type)) &&
+      (!GetContext().allowedOnceClauses.empty() &&
+          !GetContext().allowedOnceClauses.test(type))) {
```

Comment:
Why do you need the two calls to `empty()` in this condition?

Reply:
The first `allowedClauses.empty()' call is not necessary but the second one is necessary because `allowedOnceClauses` can be empty.

Reply:
Addressed in [f10920a](f10920a83170b9350b7efd8a164f9b0f70ac46dc).

Reply:
This was addressed in commit [70cea22](https://github.com/flang-compiler/f18/pull/493/commits/70cea22d4aca58c63aa3184a5ad743f5ef025796).